### ISSUE TITLE
Internal Compiler Error: Aborting - A Small Custom Model

### ISSUE THREAD

**Issue by pirazor** [google-coral/edgetpu#273]
Hello,

I would like to kindly get your help in compiling this int8 quantized model. This is ERFNet, a segmentation model that has all the supported operations that should be compiled on Edge Tpu. I am using the latest Edge TPU Compiler version 15.0.340273435.

[Quantized tflite model](https://drive.google.com/uc?export=download&id=15WnUqpdGuPM0XZmtZbHT5sZCXmdpPo0s)

Thank you. 

![erfnet](https://user-images.githubusercontent.com/4476779/101194993-c49cbf80-362c-11eb-8516-7cfad42db217.png)

**Comment by Namburger** [comment]
@usbguru @Naveen-Dodda could you guys take a look at this model?

**Comment by pirazor** [comment]
Thanks for very fast response @Namburger !

**Comment by PINTO0309** [comment]
I also quantized ERFNet and compiled it into EdgeTPU for a test run.

Edge TPU Compiler version 15.0.340273435
TensorFlow tf-nightly==2.5.0-dev20201204
Dataset Cityscapes
The model's citation repository https://github.com/Eromera/erfnet_pytorch.git

- 256x512, 384x768, 512x1024
**https://github.com/PINTO0309/PINTO_model_zoo/tree/master/075_ERFNet**

Strangely enough, 512x1024 was aborted by the edgetpu_compiler, but the other resolutions compiled successfully.
![Screenshot 2020-12-05 20:13:07](https://user-images.githubusercontent.com/33194443/101241059-4f1d1600-3736-11eb-9ae1-64fb209de3e1.png)

<details><summary><b>Model structure diagram of ERFNet Full Integer Quant tflite</b></summary><div>

![erfnet_384x768_cityscapes_full_integer_quant tflite](https://user-images.githubusercontent.com/33194443/101241075-77a51000-3736-11eb-8737-d3d9b1c38748.png)

</div></details>

**Comment by pirazor** [comment]
Thank you @PINTO0309, amazing effort ! Did you use ONNX converter to a saved model, then implement the post-quantization ?

**Comment by PINTO0309** [comment]
@pirazor 
I converted while optimizing the model with the following flow.
**`PyTorch -> ONNX -> ONNX Simplifier -> OpenVINO -> openvino2tensorflow`**

**`openvino2tensorflow`** automatically generates TensorFlow Lite and EdgeTPU Compiler-optimized NHWC models for 4D input models in NCHW format. Also, **`openvino2tensorflow`** can automatically generate a Post-Training quantization model.

## Procedure
1. Generate ONNX files from https://github.com/Eromera/erfnet_pytorch.git
2. Optimizing the ONNX model with https://github.com/daquexian/onnx-simplifier.git
3. https://docs.openvinotoolkit.org/latest/openvino_docs_install_guides_installing_openvino_linux.html while converting the ONNX model to IR format using OpenVINO Optimization (Model Optimizer)
4. https://github.com/PINTO0309/openvino2tensorflow.git openvino2tesnsorflow is used to generate a more optimized state model. It also automatically solves the problems of the TFLite Converter or EdgeTPU compiler.

## Problems encountered by the TFLite Converter or EdgeTPU compiler
1. [issue : Resize Nearest Neighbor does not convert #187 - google-coral/edgetpu](https://github.com/google-coral/edgetpu/issues/187)
2. [issue : Yolov4-Tiny EdgeTPU Missing Leaky Relu #51 - PINTO0309/PINTO_model_zoo](https://github.com/PINTO0309/PINTO_model_zoo/issues/51)
3. [issue : RESIZE_NEAREST_NEIGHBOR Operation version not supported #42024 - tensorflow/tensorflow](https://github.com/tensorflow/tensorflow/issues/42024)

## Problems when converting a model from NCHW to NHWC
[[English] Converting PyTorch, ONNX, Caffe, and OpenVINO (NCHW) models to Tensorflow / TensorflowLite (NHWC) in a snap](https://qiita.com/PINTO/items/ed06e03eb5c007c2e102)

**Comment by pirazor** [comment]
@PINTO0309 oh wow, that's awesome ! The struggles we are going through just to be able to serve a relatively simple model... I will give it a try. @Namburger don't get me wrong we love our Edge TPU, I just wish it would be as easy as using Jetson for inference :)

**Comment by pirazor** [comment]
@PINTO0309, I spent my entire day to obtain the same result on my ERFNet model. Your work in the openvino2tensorflow was very impressive, yet I still have no luck with successful EdgeTPU compilation.

If it is not too much trouble for you, would you mind giving a try using your method ? I provided [here](https://drive.google.com/uc?export=download&id=1BE3ZcRQ8-uQs63J--SIbftpcOzl4gzCE) the model files (onnx, simplified-onnx, small calibration data, and a conversion script for post-quantization) 

Thank you !

**Comment by PINTO0309** [comment]
@pirazor 

In processing with NHWC, the following four Transpose OPs are unnecessary and must be removed. I will return home in nine hours to work on it and let you know the result.

Incidentally, one of the major advantages of using OpenVINO IR is that you can edit the .xml file to manually remove unwanted OPs and change the axis of the transposition. There is no need to modify the Model program at all.

![image](https://user-images.githubusercontent.com/33194443/101574312-0312ef80-3a1c-11eb-9c5d-4e7eff48cae5.png)
![image](https://user-images.githubusercontent.com/33194443/101574012-ed9dc580-3a1b-11eb-8d92-cc5ab324492a.png)
![image](https://user-images.githubusercontent.com/33194443/101574740-2342ae80-3a1c-11eb-898e-cfc22982b59b.png)

- Example of manually removing Transpose OP

![image](https://user-images.githubusercontent.com/33194443/101580748-d8299b00-3a1d-11eb-808b-eaaea8fb498c.png)

**Comment by PINTO0309** [comment]
@pirazor 
Transpose before and after Softmax is very tricky. Is it possible to remove Transpose and change Softmax axis=3 to axis=1? OpenVINO optimization is unstable.

**Comment by pirazor** [comment]
@PINTO0309, apologies for my delayed response. Would you mind trying converting the attached onnx model to edgetpu model. I removed the softmax and the uncessary transpose is now disappeared. But, I am now struggling to find a suitable Tensorflow version :( 
[ERFNet_Lane](https://drive.google.com/uc?export=download&id=1lO6DelyAryaibEoIAmPqMIaDr027HFjj)
Happy New Year btw.

**Comment by Naveen-Dodda** [comment]
@pirazor Great effort in mapping ERFNet to edgetpu. Closing this due lack of activity.
